src/snowflake/cli/plugins/object/stage/diff.py

```diff
@@ -154,12 +159,31 @@ def build_md5_map(list_stage_cursor: SnowflakeCursor) -> Dict[str, str]:
     }
 
 
-def stage_diff(local_path: Path, stage_fqn: str) -> DiffResult:
+def get_absolute_files_to_stage(relative_files_to_stage, local_path) -> List[Path]:
+    paths = []
+    for file in relative_files_to_stage:
+        absolute_path = Path(os.path.join(local_path, file))
+        if not absolute_path.exists():
+            raise FileDoesNotExistError(file)
+        elif absolute_path.is_dir():
+            raise DirectoryNotSupportedError(file)
```

Comment:
why not? It makes a lot of sense to support them for our target use case

Reply:
Just to keep the scope of this PR small. I can follow up with supporting directories as a separate PR.